CRM/Financial/BAO/FinancialItem.php

```diff
@@ -286,25 +286,25 @@ public static function checkContactPresent($contactIds, &$error) {
   }
 
   /**
-   * Get last financial item data.
+   * Get most relevant previous financial item relating to the line item.
    *
-   * @param int $entityId
+   * This function specifically excludes sales tax.
    *
-   * @param string $entityTable
+   * @param int $entityId
    *
    * @return object CRM_Core_DAO
    */
-  public static function getPreviousFinancialItem($entityId, $entityTable = 'civicrm_line_item') {
-    $queryParams = array(
-      1 => array($entityId, 'Integer'),
-      2 => array($entityTable, 'String'),
+  public static function getPreviousFinancialItem($entityId) {
+    $params = array(
+      'entity_id' => $entityId,
+      'entity_table' => 'civicrm_line_item',
```

Comment:
Okay, changed my mind. We don't have proper API support for something like Drupal Commerce or WooCommerce non-civi line items at the moment, and we'll leave the spec for that till later. Let's hardcode civicrm_line_item, and force refactoring till later when a better scope is known.